internal/fusemount/fusefs.go

```diff
@@ -36,6 +38,51 @@ type fuseFileNode struct {
 	fuseNode
 }
 
+var _ fusefs.NodeOpener = (*fuseFileNode)(nil)
```

Comment:
I'm assuming that when `NodeOpener` is implemented, then `Open` below is called and then a `Handle` is used to perform the reads, instead of `ReadAll`. Is that the case?